src/App.tsx

```diff
@@ -1,14 +1,50 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import './App.scss';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { debounce, filterTodosByStatus } from './types/Helpers/helpers';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isTodoCompleted, setIsTodoComlpeted] = useState('all');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(allTodos => setTodos(allTodos))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  let visibleTodos = [...todos];
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 500),
+    [],
+  );
+
+  useMemo(() => {
+    visibleTodos = visibleTodos.filter(todo => todo.title.toLowerCase().includes(appliedQuery));
+  }, [appliedQuery]);
+
+  if (isTodoCompleted === 'completed') {
+    visibleTodos = filterTodosByStatus(visibleTodos, true);
+  }
+
+  if (isTodoCompleted === 'active') {
+    visibleTodos = filterTodosByStatus(visibleTodos, false);
+  }
```

Comment:
Let's unite this logic
1. Create a separate helper which will take an array, filter type, query, and return the filtered array.
It will looks similar to :
2. Create an enum with fillter types and use them here.

```
const filterTodos = (todos: Todo[], selectedOption: string, searchQuery: string) => {
  const preparedQuery = searchQuery.toLowerCase().trim();

  return todos.filter(todo => {
     const isTodoIncluded = todo.title.toLowerCase().includes(preparedQuery);
     
     switch(selectedOption) {
       case FilterType.All:
         return isTodoIncluded;

       case FilterType.Active:
         ....
     }
 }
```

Reply:
👌fixed